csharp/src/Ice/ObjectAdapterI.cs

```diff
@@ -1017,6 +1016,18 @@ public int messageSizeMax()
 
     internal Object dispatchPipeline { get; }
 
+    internal static void checkIdentity(Identity ident)
+    {
+        if (ident.name == null || ident.name.Length == 0)
+        {
+            throw new IllegalIdentityException(ident);
+        }
+        if (ident.category == null)
+        {
+            ident.category = "";
```

Comment:
It is odd for checkIdentity to change the identity.

Reply:
I did not change this code, just moved it after making it internal.

All these null checks for Identity are very odd because it's actually hard to create an Identity will null fields:
```csharp
public sealed partial class Identity : global::System.IEquatable<Identity>
    {
        [global::System.CodeDom.Compiler.GeneratedCodeAttribute("slice2cs", "3.8.0-alpha.0")]
        public string name = "";

        [global::System.CodeDom.Compiler.GeneratedCodeAttribute("slice2cs", "3.8.0-alpha.0")]
        public string category = "";
```

We should fix all this identity / null code is a follow-up PR.